RecentActivity/src/org/sleuthkit/autopsy/recentactivity/DomainCategoryRunner.java

```diff
@@ -0,0 +1,396 @@
+            logger.log(Level.SEVERE, "Encountered error retrieving artifacts for messaging domains", e); //NON-NLS
+        } finally {
+            if (context.dataSourceIngestIsCancelled()) {
+                logger.info("Operation terminated by user."); //NON-NLS
+            }
+            logger.log(Level.INFO, String.format("Extracted %s distinct messaging domain(s) from the blackboard.  "
+                    + "Of the %s artifact(s) with valid hosts, %s url(s) contained messaging domain suffix.",
+                    hostSuffixesSeen.size(), artifactsAnalyzed, domainTypeInstancesFound));
+        }
+    }
+
+    /**
+     * Adds a TSK_WEB_CATEGORIZATION artifact for the given information.
+     *
+     * @param artHost Pertinent details for the artifact (i.e. host, domain,
+     * parent file).
+     * @param domainCategory The category for this host/domain.
+     */
+    private void addCategoryArtifact(ArtifactHost artHost, String domainCategory) {
+        String moduleName = Bundle.DomainCategoryRunner_parentModuleName();
+        Collection<BlackboardAttribute> bbattributes = Arrays.asList(
+                new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_DOMAIN, moduleName, artHost.getDomain()),
+                new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_HOST, moduleName, artHost.getHost()),
+                new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_NAME, moduleName, domainCategory)
+        );
+        postArtifact(createArtifactWithAttributes(ARTIFACT_TYPE.TSK_WEB_CATEGORIZATION, artHost.getAbstractFile(), bbattributes));
+    }
+
+    @Override
+    public void process(Content dataSource, IngestJobContext context, DataSourceIngestModuleProgress progressBar) {
+        this.dataSource = dataSource;
+        this.context = context;
+
+        progressBar.progress(Bundle.DomainCategoryRunner_Progress_Message_Domain_Types());
+        this.findDomainTypes();
+    }
+
+    @Override
+    void configExtractor() throws IngestModule.IngestModuleException {
+        // lookup all providers, filter null providers, and sort providers
+        List<DomainCategorizer> foundProviders
+                = Lookup.getDefault().lookupAll(DomainCategorizer.class).stream()
+                        .filter(provider -> provider != null)
+                        .sorted((a, b) -> a.getClass().getName().compareToIgnoreCase(b.getClass().getName()))
+                        .collect(Collectors.toList());
+
+        // add the default categorizer last as a last resort
+        foundProviders.add(new DefaultDomainCategorizer());
+
+        for (DomainCategorizer provider : foundProviders) {
+            try {
+                provider.initialize();
+            } catch (DomainCategorizerException ex) {
+                throw new IngestModule.IngestModuleException("There was an error instantiating the provider: " + provider.getClass().getSimpleName(), ex);
+            }
+        }
+
+        this.domainProviders = foundProviders == null
```

Comment:
Can foundProviders be null? I don't think so, but even if it could, you would get an NPE on line 367.